extension/cocoatable.js

```diff
@@ -145,13 +145,25 @@
 			nextCell.open();
 		}
 	}
+	CocoaTable.prototype.onSelectedRowChanged = function (callback) {
+		// FUTURE add cleanup of previous listener
+		return this._table.addEventListener('selectedRowChanged', callback);
+	}
 	CocoaTable.prototype.setSelectedRow = function (row) {
+		var prevRow = this._selectedRow;
 		if ( this._selectedRow ) {
 			removeClass(this._selectedRow, 'selected');
 		}
 		if ( row )
 			addClass(row, 'selected');
 		this._selectedRow = row;
+		var event = new CustomEvent('selectedRowChanged', {
+			'detail': {
+				'previous': prevRow,
+				'current': this._selectedRow
+			}
+		});
+		this._table.dispatchEvent(event);
 	}
```

Comment:
I chose to add this new event so that we could be notified when the row selection changed and then use that info to update the button text for changing the default format. At first, I looked into adding a new column, "Default", that acted like a radiobutton, but that quickly got very ugly and complicated :)